modules/store/src/store.ts

```diff
@@ -93,6 +94,17 @@ export class Store<T = object>
     return (select as any).call(null, pathOrMapFn, ...paths)(this);
   }
 
+  /**
+   * Returns a signal of the provided selector.
+   * This method must be run within in injection context.
+   *
+   * @param selectorFn selector function
+   */
+  selectSignal<K>(selectorFn: (state: T) => K): Signal<K>;
+  selectSignal<K = unknown>(selectorFn: (state: T) => K): Signal<unknown> {
+    return toSignal(this.select(selectorFn), { requireSync: true });
+  }
```

Comment:
What do you think about the following implementation:

```diff
@Injectable()
export class Store<T = object>
  extends Observable<T>
  implements Observer<Action>
{
+  private readonly state: Signal<T>;

  constructor(
    state$: StateObservable,
    private actionsObserver: ActionsSubject,
    private reducerManager: ReducerManager,
+    private injector: Injector
  ) {
    super();

    this.source = state$;
+    this.state = runInInjectionContext(injector, () =>
+      toSignal(state$ as Observable<T>, { requireSync: true })
+    );
  }

+  selectSignal<R>(selector: (state: T) => R): Signal<R> {
+    return computed(() => selector(this.state()));
+  }
}
```

It would not require the `selectSignal` method to be called within the injection context.

Reply:
Yea, I tried this option also.  It is better that it doesn't need to be called within the injection context.

It makes it a breaking change because of the added dependency on the `Injector` and breaks some tests where we are creating a `new Store()` manually. We also have to update the `Store.lift` method.

This may be the same pattern we end up using with ComponentStore also